mex/extractors/igs/transform.py

```
@@ -113,3 +129,168 @@ def transform_igs_access_platform(
+    igs_schemas: dict[str, IGSSchema],
+    extracted_igs_resource_ids_by_identifier_in_primary_source: dict[
+        str, MergedResourceIdentifier
+    ],
+    extracted_primary_source_igs: ExtractedPrimarySource,
+    extracted_igs_variable_group_ids_by_igs_identifier: dict[
+        str, MergedVariableGroupIdentifier
+    ],
+    variable_mapping: VariableMapping,
+    variable_pathogen_mapping: VariableMapping,
+    igs_info: IGSInfo,
+) -> list[ExtractedVariable]:
+    """Transform igs schemas to variables.
+
+    Args:
+        igs_schemas: igs schemas by schema name
+        extracted_igs_resource_ids_by_identifier_in_primary_source:
+            igs resources by pathogen
+        extracted_primary_source_igs: extracted primary source igs
+        extracted_igs_variable_group_ids_by_igs_identifier:
+            igs variable group by identifier in primary source
+        variable_mapping: variable mapping default values
+        variable_pathogen_mapping: variable_pathogen mapping default values
+        igs_info: igs info
+
+    Returns:
+        list of ExtractedVariable
+    """
+    description_by_enum = {
+        rule.forValues[0]: rule.setValues
+        for rule in variable_pathogen_mapping.description[0].mappingRules
+        if rule.forValues
+    }
+    enums_by_property_name = get_enums_by_property_name(igs_schemas)
+    extracted_variables: list[ExtractedVariable] = []
+    used_in = extracted_igs_resource_ids_by_identifier_in_primary_source[
+        f"IGS_{igs_info.title}_v{igs_info.version}"
+    ]
+    for schema_name, schema in igs_schemas.items():
+        data_type: str | None = None
+        if schema_name == "Pathogen" and isinstance(schema, IGSEnumSchema):
+            for enum in schema.enum:
+                belongs_to = (
+                    [extracted_igs_variable_group_ids_by_igs_identifier[schema_name]]
+                    if schema_name in extracted_igs_variable_group_ids_by_igs_identifier
+                    else []
```

Wouldn't it be better to let the extractor fail if the group is missing? Maybe mdmgmt has an opinion about that?

---

Added a comment in the wrap up ticket